### DISCUSSION TITLE
Feature request: option to freeze slug per section or disable permission for users to edit slugs

### DISCUSSION THREAD

**Discussion by TomDeSmet** [craftcms/cms#8594] [Ideas]
This feature would be nice as sometimes you may need to hard-code using a specific section, category or entry. There are multiple examples but the simplest one would be adding a section ID to a front-end save entry form.

The reason why using an ID is not a good option here, is because when adding new features to a website which is also in production, the edits on the live website continue, but you're developing on your development server.
In the end, you migrate (manually or by using some scripts) the changes from the dev to the live server but ID's may differ (maybe you targeted a spcific entry in a section, but that section may now have gained extra entries which makes the ID different). 
So bypassing that, you can get the ID of the element via its slug, which doesn't change between environments. But it would be safer to have an option that a user cannot edit that slug, only a full admin would be able to do that.

I know there are multiple ways to get around this problem in a lot of circumstances and it's always best to go that route, but that's not always possible. Hence my feature request.

**Comment by brandonkelly** [comment]
We should do this alongside #947.

**Reply by brandonkelly** [reply]
Implemented #8154 (formerly #947) differently than originally expected, so nevermind.

**Comment by samuel-clara** [comment]
Is this now possible?

**Reply by brandonkelly** [reply]
No, sorry.

**Comment by pc-erin** [comment]
This would be really useful for a client of ours. There's a plugin that's supposed to do this, but it's out of date and no longer maintained.

**Comment by brandonkelly** [comment]
Craft 5 is adding a new “Show the Slug field” setting, which should help here. (#13799)
